src/main/java/io/fabric8/maven/docker/assembly/DockerFileBuilder.java

```diff
@@ -195,28 +195,20 @@ private static void buildOption(StringBuilder b, DockerFileOption option, Object
 
     private void addCopy(StringBuilder b) {
         if (assemblyUser != null) {
-            String tmpDir = createTempDir();
-            addCopyEntries(b, tmpDir);
-
-            String[] userParts = StringUtils.split(assemblyUser, ":");
-            String userArg = userParts.length > 1 ? userParts[0] + ":" + userParts[1] : userParts[0];
-            String chmod = "chown -R " + userArg + " " + tmpDir + " && cp -rp " + tmpDir + "/* / && rm -rf " + tmpDir;
-            if (userParts.length > 2) {
-                DockerFileKeyword.USER.addTo(b, "root");
-                DockerFileKeyword.RUN.addTo(b, chmod);
-                DockerFileKeyword.USER.addTo(b, userParts[2]);
-            } else {
-                DockerFileKeyword.RUN.addTo(b, chmod);
+            String[] userParts = assemblyUser.split(":");
+
+            for (CopyEntry entry : copyEntries) {
+                String dest = (basedir.equals("/") ? "" : basedir) + "/" + entry.destination;
+                DockerFileKeyword.ADD.addTo(b, " --chown="
+                        + (userParts.length > 2 ?
```

Comment:
Really > 2 ? Not > 1 ?

Reply:
I'm asking because I don't see `userPart[2]` used anywhere.

Reply:
I think, for backwards compatibility we shoudl use `userParts[2]` to switch to that user after the `ADD`, like in the original code. (but we can mark that as deprecated, as its not really useful anymore IMO)

Reply:
Sorry, I might have missed this. Let me remove this and add a deprecation note to doc on next update